src/PrestaShopBundle/Form/Admin/Type/DownloadableFileType.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type;
+
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\HttpFoundation\File\File;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class DownloadableFileType extends CommonAbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('file', FileType::class, $options['file_options'])
+            ->add('download_file_url', HiddenType::class)
+        ;
+
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $formEvent) {
+            $this->handleDownloadFileVisibility($formEvent);
+        });
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefault('file_options', []);
+        $resolver->addAllowedTypes('file_options', 'array');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBlockPrefix()
+    {
+        return 'downloadable_file';
+    }
+
+    /**
+     * @param FormEvent $formEvent
+     */
+    private function handleDownloadFileVisibility(FormEvent $formEvent): void
+    {
+        $data = $formEvent->getData();
+
+        if (!$data || !$data['file'] instanceof File) {
+            $formEvent->getForm()->remove('download_file');
```

Comment:
I'm not sure to understand this one 🤔 I don't see any `download_file` form field in the PR, so I wonder which field this is referring to. Is the prupose to remove this `DownloadableFileType` field or something else?

Also the instanceof check seems like it would be safer with some parenthesis

```php
            if (!$data || !($data['file'] instanceof File)) {
```

Reply:
yeah i've changed it and didn't correct. It supposed to be 'download_file_url' whcich is the filed in DownloadableFileType that should only appear if file is provided